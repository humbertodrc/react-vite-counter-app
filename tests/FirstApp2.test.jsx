import {render, screen} from "@testing-library/react";
import {FirstApp} from "../src/FirstApp";

describe("Pruebas en <FirstApp /> dos", () => {
	const title = "Hola soy un Titulo";
	const subTitle = "Hola soy un subtitulo";

	test("Debe de hacer match con el snapshot", () => {
		const {container} = render(<FirstApp title={title} />);

		expect(container).toMatchSnapshot();
	});

	test('Debe mostar el mensaje "Hola soy un Titulo"', () => {
		render(<FirstApp title={title} />);
		// screen.debug()
		expect(screen.getByText(title)).toBeTruthy();
	});

	test('Debe mostrar el titulo en un h1', () => {
		render(<FirstApp title={title} />);
		expect(screen.getByRole('heading', {leve: 1 }).innerHTML).toContain(title)
	})

	test('Debe mostart el subtitulo enviado por props', () => {
		render(<FirstApp title={title} subTitle={subTitle} />);

		expect(screen.getAllByText(subTitle).length).toBe(2)
	})
});
