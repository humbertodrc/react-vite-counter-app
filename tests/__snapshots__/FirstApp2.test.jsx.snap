// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> dos Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy un Titulo
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Humberto Rivero
  </p>
</div>
`;
